lib/sort_by_length.rb

```diff
@@ -1,7 +1,36 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
 # Time complexity: ?
+# to why it was not => nlogn
+# i   j
+# 1   1
+# 2   2
+# 3   3
+# 4   4
+# 5   5
+# .   .
+# .   .
+# n   n
+
+# n + n(n+1)/2 = n +  (n2 + n)/2  => O(n2)
+
 # Space complexity: ?
+# n => array
+# O(n)
+
 def sort_by_length(my_sentence)
```

Comment:
Good implementation of Insertion sort.  The time complexity is O(n^2) since you have the two nested loops, and by creating the array you have O(n) space complexity.